### PR TITLE
fix: [OMV-481] DRYs up code and matches mdformat output

### PR DESCRIPTION
**Pull request by BenjaminNolan**
## Summary by CodeRabbit

- **New Features**
	- Enhanced documentation generation with automatic table of contents and improved table formatting for services, messages, and enums.
- **Refactor**
	- Improved code readability and maintainability across various internal components.
	- Centralized table writing functionality using utility functions.
- **Tests**
	- Extended test coverage to include new functionalities and utility methods.

### CODE REVIEW

**Comment by BenjaminNolan** [comment]
@coderabbitai review

File: internal/converter/converter.go [hunk 1]
```diff
@@ -98,12 +100,16 @@ func (c *converter) addFile(file *protogen.File) {
 	for _, svc := range file.Services {
 		c.packages[pkgName].services[string(svc.Desc.Name())] = svc
 	}
-}
 
-func (c *converter) processFile(plugin *protogen.Plugin, file *protogen.File) error {
-	c.addFile(file)
+	g := pl.NewGeneratedFile(file.GeneratedFilenamePrefix+suffix, file.GoImportPath)
```
Comment (lplejic-ogcs) [inline internal/converter/converter.go:104]: Can you rename `g` to something else for better readability

File: internal/converter/converter_test.go [hunk 2]
```diff
@@ -1,15 +1,92 @@
...
+	}
+
+	return c, nil
+}
+
+func (ts *ConverterTestSuite) TestConverter() {
+	c, err := getTestConverter()
+	ts.NoError(err)
+
+	pl, err := getTestPlugin()
```
Comment (lplejic-ogcs) [inline internal/converter/converter_test.go]: same above for `c` and `pl`

**Review by asahm-omnipeak — DISMISSED** [review]

File: internal/converter/converter.go (hunk 1 above)
Comment (BenjaminNolan) [reply internal/converter/converter.go:104]: `g` is the convention for the usage of `protogen.GeneratedFile` in protoc plugins:
* https://github.com/search?q=g+protogen.GeneratedFile&type=code
* https://github.com/search?q=repo%3Afischor%2Fprotogen-python%20protogen.GeneratedFile&type=code

**Review by lplejic-ogcs — APPROVED** [review]

**Pull request merged**